src/main/java/mServer/crawler/sender/MediathekArd.java

```diff
@@ -337,125 +326,139 @@ private void filmSuchen2(String urlFilm_, String thema, String titel, long dauer
-                    }
-                    if (urlHD.contains("?url=")) {
-                        urlHD = url.substring(0, urlHD.indexOf("?url="));
-                    }
+				// No Url found? Try M3U8 / HLS:
+				if (isAllTextsEmpty(url, urlHD, urlMid, urlKl)) {
+					Map<Qualities, String> urls = searchForUrlsWithM3U8(seite2);
+					if (urls.containsKey(Qualities.SMALL)) {
+						urlKl = urls.get(Qualities.SMALL);
+					}
+					if (urls.containsKey(Qualities.NORMAL)) {
+						urlMid = urls.get(Qualities.NORMAL);
+					}
+					if (urls.containsKey(Qualities.HD)) {
+						urlHD = urls.get(Qualities.HD);
+					}
+				}
 
-                    String beschreibung = beschreibung(urlSendung);
-                    DatenFilm f = new DatenFilm(SENDERNAME, thema, urlSendung, titel, url, ""/*urlRtmp*/, datum, zeit, dauer, beschreibung);
-                    if (!urlKl.isEmpty()) {
-                        CrawlerTool.addUrlKlein(f, urlKl, "");
-                    }
-                    if (!urlHD.isEmpty() && !urlHD.equals(url)) {
-                        CrawlerTool.addUrlHd(f, urlHD, "");
-                    }
-                    if (!subtitle.isEmpty()) {
-                        CrawlerTool.addUrlSubtitle(f, subtitle);
-                    }
-                    addFilm(f);
-                } else {
-                    filmSuchen_old(urlSendung, thema, titel, dauer, datum, zeit);
-//                    MSLog.fehlerMeldung(784512369, "keine URL: " + urlFilm);
-                }
-            } catch (Exception ex) {
-                Log.errorLog(762139874, ex);
-            }
-        }
+				if (!urlTest.isEmpty() && (urlTest.equals(url) || urlTest.equals(urlKl))) {
+					urlHD = ""; // dann ists kein HD
+					// System.out.println("q3 test: " + urlFilm);
+				}
+				if (!urlTest.isEmpty() && !(urlTest.equals(url) || urlTest.equals(urlMid))) {
+					// System.out.println("q3 test: " + urlFilm);
+				}
+				if (url.isEmpty() && urlMid.isEmpty() && urlKl.isEmpty() && !thema.equals(THEMA_ALPHA_CENTAURI)) {
+					// System.out.println("q3 test: " + urlFilm);
+				}
+				if (urlMid.isEmpty()) {
+					// System.out.println("q2test: " + urlFilm);
+				}
+				if (urlKl.isEmpty()) {
+					// System.out.println("q1 test: " + urlFilm);
+				}
+				if (url.isEmpty()) {
+					url = urlMid;
+					urlMid = "";
+				}
+				if (url.isEmpty()) {
+					url = urlKl;
+					urlKl = "";
+				}
+				if (url.isEmpty() && !urlHD.isEmpty()) {
+					url = urlHD;
+					urlHD = "";
+				}
+				if (urlKl.isEmpty()) {
+					urlKl = urlMid;
+				}
+				String subtitle = seite2.extract("subtitleUrl\":\"", "\"");
+				if (!subtitle.isEmpty()) {
+					if (!subtitle.startsWith("http://www.ardmediathek.de")) {
+						subtitle = "http://www.ardmediathek.de" + subtitle;
+					}
+				}
+				if (!url.isEmpty()) {
+
+					// http://http-stream.rbb-online.de/rbb/rbbreporter/rbbreporter_20151125_solange_ich_tanze_lebe_ich_WEB_L_16_9_960x544.mp4?url=5
+					if (url.contains("?url=")) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "?url=" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)